drivers/cpufreq/cpu_input_boost.c

```diff
@@ -241,6 +251,12 @@ static void cpu_ib_input_event(struct input_handle *handle, unsigned int type,
 	enum boost_status ib_status;
 	bool do_boost;
 
+	/* Anticipate device wake-up and boost early */
```

Comment:
Boosting while the screen is off caused panics and cpufreq violations for me with MSM8974 kernels. This didn't crash my MSM8996 kernel, but it still resulted in cpufreq violations, which is why I've stopped using it entirely.